server/game/gameSystems/PlayCardSystem.ts

```diff
@@ -0,0 +1,81 @@
+import type { Card } from '../core/card/Card';
+import AbilityResolver from '../core/gameSteps/AbilityResolver';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import * as Contract from '../core/utils/Contract';
+import { CardType, PlayType } from '../core/Constants';
+import { isPlayable } from '../core/card/CardTypes';
+import * as GameSystemLibrary from './GameSystemLibrary';
+import { PlayCardAction } from '../core/ability/PlayCardAction';
+import { PlayUnitAction } from '../actions/PlayUnitAction';
+import { PlayUpgradeAction } from '../actions/PlayUpgradeAction';
+import { PlayEventAction } from '../actions/PlayEventAction';
+
+export interface IPlayCardProperties extends ICardTargetSystemProperties {
```

Comment:
It's kind of an edge case but there is a missing feature here we need to add a flag for. Specifically for the Mon Mothma example in the rule below (this is 1.17.4):

![image](https://github.com/user-attachments/assets/065ac5b8-4947-4540-8489-8216df3d1876)

Reply:
I haven't thought through what the best way to do this is. I'm not sure if it can be handled automatically or if we need to just have a flag that like "passHiddenInformation" or something. And to avoid a bunch of spurious bug reports, probably the prompt itself should explain _why_ the player is being presented with a pass option even though the ability doesn't say "you may."

Reply:
I changed CardTargetResolver to always set itself to optional if it's only targeting cards in hidden zones. I also figured out how to edit the prompt for the player to inform them like you suggested(it was non-obvious because I had to mess with the `selector` member of the CardTargetResolver).

Reply:
Awesome!! Great work